x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -880,8 +1283,8 @@ LogEntryBuilder with(Map<String, Object> map) {
             return this;
         }
 
-        StringMapMessage build() {
-            return logEntry;
+        void build() {
+            logger.info(AUDIT_MARKER, logEntry);
```

Comment:
Well, this is part of what we're debating at https://github.com/elastic/elasticsearch/pull/62916#discussion_r537439183 .
If we wish to abstract the log entry building we cannot keep exposing the logger, because the logger expects certain fields that the entry builder populates (the logger has a configuration attached from its log4j2.properties file).

Would renaming `build` to `log` satisfy you?

Reply:
No need to rename because this is just a Nit. But I don't understand what do you mean by "If we wish to abstract the log entry building we cannot keep exposing the logger". Could you please elaborate?

Reply:
> I don't understand what do you mean by "If we wish to abstract the log entry building we cannot keep exposing the logger". Could you please elaborate?

The `LogEntryBuilder` builds a `StringMapMessage`. The keys in the `MapMessage` must match the keys in the `log4j2.properties` layout pattern that is assigned to the `logger` for the `LoggingAuditTrail` class.
So I don't want to expose (make it non private) the `logger` because the caller doesn't know how to use it. I prefer we expose the `LogEntryBuilder`, which encapsulates the `logger`, because it has methods in its interface. This way, the keys from the log4j pattern stay hidden from the callers.